test/common/wpt.js

```diff
@@ -587,6 +683,8 @@ class WPTRunner {
         }
       }
 
+      this.report?.write();
```

Comment:
Can you add a comment to here for explaining the conditional case here. It would be easier to understand & trace. (for me)